pygeoapi/util.py

```diff
@@ -886,3 +879,131 @@ def bbox2geojsongeometry(bbox: list) -> dict:
+    actual name of the provider's geometry field
+    :returns: A new pygeofilter.ast.Node, with the modified filter
+    expression
+
+    This function modifies the parsed pygeofilter that contains the raw
+    filter expression provided by an external client. It performs the
+    following modifications:
+
+    - if the filter includes any spatial coordinates and they are being
+      provided in a different CRS from the provider's storage CRS, the
+      corresponding geometries are transformed into the storage CRS
+
+    - if the filter includes the generic 'geometry' name as a reference to
+      the actual geometry of features, it is replaced by the actual name
+      of the geometry field, as specified by the provider
+
+    """
+    new_tree = deepcopy(ast_tree)
+    if storage_crs_uri:
+        storage_crs = get_crs_from_uri(storage_crs_uri)
+        filter_crs = get_crs_from_uri(filter_crs_uri)
+        _inplace_transform_filter_geometries(new_tree, filter_crs, storage_crs)
+    if geometry_column_name:
+        _inplace_replace_geometry_filter_name(new_tree, geometry_column_name)
+    return new_tree
+
+
+def _inplace_transform_filter_geometries(
+        node: pygeofilter.ast.Node,
+        filter_crs: pyproj.CRS,
+        storage_crs: pyproj.CRS
+):
+    """
+    Recursively traverse node tree and convert coordinates to the storage CRS.
+
+    This function modifies nodes in the already-parsed filter in order to find
+    any geometry literals that may be used in the filter and, if necessary,
+    proceeds to convert spatial coordinates to the CRS used by the provider.
+    """
+    try:
+        sub_nodes = node.get_sub_nodes()
+    except AttributeError:
+        pass
+    else:
+        for sub_node in sub_nodes:
+            is_geometry_node = isinstance(
+                sub_node, pygeofilter.values.Geometry)
+            if is_geometry_node:
+                # NOTE1: To be flexible, and since pygeofilter
+                # already supports it, in addition to supporting
+                # the `filter-crs` parameter, we also support having a
+                # geometry defined in EWKT, meaning the CRS is provided
+                # inline, like this `SRID=<CRS_CODE>;<WKT>` - If provided,
+                # this overrides the value of `filter-crs`. This enables
+                # supporting, for example, an exotic filter expression with
+                # multiple geometries specified in different CRSs
+
+                # NOTE2: We specify a default CRS using a URI of type URN
+                # because this is what pygeofilter uses internally too
+
+                crs_urn_provided_in_ewkt = sub_node.geometry.get(
+                    'crs', {}).get('properties', {}).get('name')
+                if crs_urn_provided_in_ewkt is not None:
+                    crs = get_crs_from_uri(crs_urn_provided_in_ewkt)
+                else:
+                    crs = filter_crs
+                print(f"{crs=}")
```

Comment:
Can you use the logger instead of the print function?

Reply:
Those were not needed, so I removed them